[fx] get rid of graph_module.root

zdevito opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44092 [fx] get rid of graph_module.root**

instead submodules and weights are installed directly on the
graph_module by transferring the original modules. This makes it more
likely that scripting will succeed (since we no longer have submodules
that are not used in the trace). It also prevents layered transforms
from having to special case handling of the `root` module. GraphModules
can now be re-traced as part of the input to other transforms.

Differential Revision: [D23504210](https://our.internmc.facebook.com/intern/diff/D23504210)

zdevito committed:
[fx] get rid of graph_module.root

instead submodules and weights are installed directly on the
graph_module by transferring the original modules. This makes it more
likely that scripting will succeed (since we no longer have submodules
that are not used in the trace). It also prevents layered transforms
from having to special case handling of the `root` module. GraphModules
can now be re-traced as part of the input to other transforms.

[ghstack-poisoned]

zdevito committed:
Update on "[fx] get rid of graph_module.root"


instead submodules and weights are installed directly on the
graph_module by transferring the original modules. This makes it more
likely that scripting will succeed (since we no longer have submodules
that are not used in the trace). It also prevents layered transforms
from having to special case handling of the `root` module. GraphModules
can now be re-traced as part of the input to other transforms.

[ghstack-poisoned]

jamesr66a commented on torch/fx/graph_module.py:
```diff
@@ -34,7 +35,7 @@ def forward_from_src(src : str):
+def deserialize_graphmodule(body : dict) -> torch.nn.Module:
     """
     Deserialize a GraphModule given the original `root` module and the generated
     `forward()` source code (`src`). This will exec() the source of the forward
```
Can you update the comment here?

jamesr66a reviewed (approved):
![image](https://user-images.githubusercontent.com/4685384/92078628-7220de00-ed73-11ea-8b01-b6929944e974.png)

zdevito committed:
Update on "[fx] get rid of graph_module.root"


instead submodules and weights are installed directly on the
graph_module by transferring the original modules. This makes it more
likely that scripting will succeed (since we no longer have submodules
that are not used in the trace). It also prevents layered transforms
from having to special case handling of the `root` module. GraphModules
can now be re-traced as part of the input to other transforms.

[ghstack-poisoned]
